src/main/java/com/shipmentEvents/handlers/EventHandler.java

```diff
@@ -67,7 +67,9 @@ public String weakMessageEncryption(String message, String key) throws Exception
         Cipher cipher = Cipher.getInstance("RSA");
         SecretKey secretKey = new SecretKeySpec(key.getBytes(), "AES");
         cipher.init(Cipher.ENCRYPT_MODE, secretKey);
-        return new String(cipher.doFinal(message.getBytes()), StandardCharsets.UTF_8);
+        String test = new String(cipher.doFinal(message.getBytes()), StandardCharsets.UTF_8);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### String.getBytes [](|b0bf3823-2844-464e-a2b1-b7080947ddc2|)

Using this method without an explicit charset/encoding causes the default character encoding of the JVM to be used for conversion from Java’s internal Unicode encoding to bytes. The system’s default encoding, which is used to determine the bytes for unsafe characters, might not always give correct results.

#### Solution:
Specify an encoding

*(likely `UTF-8`, which is backward compatible with ASCII and has multi-language support)*

```
byte[] messageBytes = message.getBytes(StandardCharsets.UTF_8);
```